src/main/kotlin/org/partiql/scribe/targets/trino/TrinoRewriter.kt

```diff
@@ -162,8 +162,8 @@ public open class TrinoRewriter(val onProblem: ProblemCallback) : PlanRewriter<R
     }
 
     override fun visitRexOpPathSymbol(node: Rex.Op.Path.Symbol, ctx: Rel.Type?): PlanNode {
-        if (node.root.op !is Rex.Op.Var) {
-            error("Trino does not support path expressions on non-variable values")
+        if (node.root.type.asNonAbsent() !is StructType) {
+            error("Trino path expression must be on a ROW type (PartiQL STRUCT), found ${node.root.type}")
```

Comment:
Main logic change in this PR. Changed to use the same assertion that we used for the above `visitRexOpPathKey`. I believe the logic change from https://github.com/partiql/partiql-scribe/commit/0202775650c878a33af714bbe8e58320c3da69da#diff-e60aff0e7d40622acdaef39c8765f8d91fa61a064946fbb227127f735a825e3d should have also been applied to `visitRexOpPathSymbol`.